packages/flutter_tools/test/integration/test_driver.dart

```diff
@@ -42,6 +36,17 @@ class FlutterTestDriver {
   VMServiceClient vmService;
   String get lastErrorInfo => _errorBuffer.toString();
 
+  String _debugPrint(String msg) {
+    const int maxLength = 500;
+    final String truncatedMsg =
+        msg.length > maxLength ? msg.substring(0, maxLength) + '...' : msg;
+    _allMessages.add(truncatedMsg);
+    if (_printJsonAndStderr) {
+      print(truncatedMsg);
```

Comment:
You could also consider using the `print` variant from package:test that only prints when the test fails.

Reply:
gotcha